addons/l10n_mn/migrations/1.1/post-migrate.py

```diff
@@ -0,0 +1,259 @@
+        'vat_report_tag28': 'TT3a[27]',
+        'vat_report_tag29': 'TT3a[28]',
+        'vat_report_tag33': 'TT3a[32]',
+        'vat_report_tag35': 'TT3a[34]',
+        'vat_report_tag36': 'TT3a[35]',
+        'vat_report_tag37': 'TT3a[36]',
+        'vat_report_tag38': 'TT3a[41]',
+        'vat_report_tag41': 'TT3a[44]',
+        'vat_report_tag42': 'TT3a[45]',
+        'vat_report_tag43': 'TT3a[38]',
+        'vat_report_tag44': 'TT3a[46]',
+        'vat_report_tag45': 'TT3a[40]',
+        'vat_report_tag46': 'TT3a[48]',
+        'vat_report_tag48': 'TT3a[50]',
+        'vat_report_tag49': 'TT3a[51]',
+        'vat_report_tag51': 'TT3a[53]',
+        'vat_report_tag52': 'TT3a[54]',
+    }
+    # 3. The VAT Report has been converted to use the tax tags engine.
+    #
+    # 3.1. Previously, some tax tags used to be set on both base lines and tax lines,
+    #   and were distinguished in the report using domains like ('tax_line_id', '!=', False)
+    #   or ('tax_id', '!=', False).
+    #
+    #   Because the tax tags engine doesn't discriminate between tax lines and base lines,
+    #   the tax tags that are meant for tax lines need to be unset on existing base lines,
+    #   and the tax tags that are meant for base lines need to be unset on existing tax lines.
+    sale_vat_tags = [
+        'vat_report_tag2',
+        'vat_report_tag5',
+        'vat_report_tag6',
+        'vat_report_tag7',
+        'vat_report_tag8',
+        'vat_report_tag9',
+        'vat_report_tag11',
+        'vat_report_tag12',
+        'vat_report_tag13',
+        'vat_report_tag14',
+        'vat_report_tag15',
+        'vat_report_tag16',
+        'vat_report_tag17',
+        'vat_report_tag18',
+        'vat_report_tag19',
+        'vat_report_tag20',
+        'vat_report_tag21',
+        'vat_report_tag22',
+        'vat_report_tag23',
+        'vat_report_tag24',
+        'vat_report_tag25',
+        'vat_report_tag28',
+        'vat_report_tag29',
+        'vat_report_tag48',
+        'vat_report_tag49',
+    ]
+    tax_tags_for_base_lines = sale_vat_tags + [
+        'vat_report_tag33',
+        'vat_report_tag35',
+        'vat_report_tag36',
+        'vat_report_tag37',
+        'vat_report_tag38',
+        'vat_report_tag51',
+        'vat_report_tag52',
+    ]
+    tax_tags_for_tax_lines = [
+        'vat_report_tag41',
+        'vat_report_tag42',
+        'vat_report_tag43',
+        'vat_report_tag44',
+        'vat_report_tag45',
+        'vat_report_tag46',
+    ]
+    # 3.2. In the previous version of the VAT report, some lines did a sum of AMLs
+    #   whereas others did a -sum of AMLs. The new tax tags engine does a sum of
+    #   (AML balance) * (tax tag sign) * (-1 if AML.tax_tag_invert else 1)
+    #   so, when converting tax tags on AMLs, we need to choose the tax tag sign
+    #   such that the sum stays the same.
+    minus_sum_tax_tags = sale_vat_tags
+
+    # Now, do the migration!
+    # STEP 1: Remove obsolete tag vat_report_tag11
```

Comment:
Yes, the tax tag does try to get removed automatically, but there are several reasons that made me want to remove it manually:
- the garbage-collection of no-longer-defined XMLids only happens after all modules are loaded, so if an unlink fails at that time, the modules themselves are not rolled back. This means that the database will be left in an inconsistent state (the upgrade will fail, but the modules will be upgraded, and potentially some of the data to be deleted will in fact not have been deleted). 
- if a tax tag is still referenced on an AML or a tax repartition line, unlinking it will fail due to the `ONDELETE RESTRICT`, so we would trigger precisely the above situation that we want to avoid.
- we also want to remove the tax tag from tax repartition line _templates_ because we need to re-load the taxes from templates at the end of the script (due to the changes in tax repartition line templates). If the tax tag isn't removed from the templates at this time, it will be re-added to the tax repartition lines at the end of the script, causing the same problem.